src/Rings/mpoly-graded.jl

```diff
@@ -1764,30 +2155,31 @@ end
 # Seems wasteful, but in many cases leads to faster overall computation.
 
 
-# Internal function: used only in _homogenization_via_saturation (immediately below)
-# This function returns gens(I) and possibly some more "small" elements of I
-# (obtained from 1 or more groebner bases of I).  If extra_gens_flag is true,
-# we compute one more groebner bases of I in the hope that they contain some small elements.
-function _gens_for_homog_via_sat(I::MPolyIdeal{T}, extra_gens_flag::Bool) where {T <: MPolyRingElem}
-  ##    @req  !is_zero(I)  "Ideal must be non-zero"
-  ##    @req  !is_one(I)   "Ideal must not be whole ring"
-  if isempty(gens(I))  throw("Ideal must have at least 1 generator"); end;
-  OrigR = parent(gens(I)[1])      # Since I is not zero, it has at least 1 gen.
-  # Next few lines: we adjoin some more small, redundant gens.
-  # !!HEURISTIC!!  We use 2*AveNumTerms as size limit for redundant gens we shall adjoin.
-  AveNumTerms = sum([length(f)  for f in gens(I)])/length(gens(I)) ## floating-point!
-  G = gens(I)
-  if extra_gens_flag
-    # compute DegRevLex GB (?and maybe DegLex GB?)
-    assume_gb_is_cached = groebner_basis(I; ordering=degrevlex(OrigR))
-    #??Good idea??        assume_gb_is_cached = groebner_basis(I; ordering=deglex(OrigR))
-  end
-  for GB in values(I.gb)
-    extra_gens = filter(f -> (length(f) < 2*AveNumTerms), elements(GB))
-    G = vcat(G,  extra_gens)
-  end
-  return G
-end
+# 2024-02-06 COMMENTED OUT:  code uplicated above (around line 1700)
+# # Internal function: used only in _homogenization_via_saturation (immediately below)
+# # This function returns gens(I) and possibly some more "small" elements of I
+# # (obtained from 1 or more groebner bases of I).  If extra_gens_flag is true,
+# # we compute one more groebner bases of I in the hope that they contain some small elements.
```

Comment:
Can this also be deleted, then?

Reply:
Probably.  As I wrote above, the main aim yesterday was to produce a draft PR so that others could check what I've done.  I am well aware that considerable cleaning must still be done.

Reply:
I have deleted some cruft.